### ISSUE TITLE
Check for proper usage of translation functions

### ISSUE THREAD

**Issue by westonruter** [WordPress-Coding-Standards/WordPress-Coding-Standards#50]
I had read that variables were discouraged from being used in the i18n functions. I assumed this was just to discourage them from being changed. Well, it's actually because some automated i18n tools will [tokenize](http://i18n.trac.wordpress.org/browser/tools/trunk/add-textdomain.php#L60) the PHP files to extract the textdomain from PHP without actually evaluating the file. See @markjaquith's blog post [Translating WordPress Plugins and Themes: Don’t Get Clever](http://markjaquith.wordpress.com/2011/10/06/translating-wordpress-plugins-and-themes-dont-get-clever/).

So a sniff should be written which makes sure that translation functions have a text domain which is a literal string. The sniff should also ensure that variables are not being used for the text to be translated, or inside of the text to be translated.

**Comment by westonruter** [comment]
I think this was accidentally closed.

**Comment by GaryJones** [comment]
Strings that span multiple lines may get highlighted as false positive e.g.:

```php
<?php esc_html_e( 'Long line
of text', 'foo' ); ?>
```

Both give "The $text arg must be a single string literal, not 'Long line of text'."

![screenshot 2016-03-23 14 35 03](https://cloud.githubusercontent.com/assets/88371/13988603/8e7640ce-f104-11e5-956b-09f3dfcab51d.png)

**Comment by westonruter** [comment]
@GaryJones interesting. PHP tokenizes that single string literal into two separate `T_CONSTANT_ENCAPSED_STRING` tokens with content `'Long line` and `of text'`

**Comment by westonruter** [comment]
It seems we need to merge two tokens of type `T_CONSTANT_ENCAPSED_STRING` that appear next to each other in `$argument_tokens`.

**Comment by GaryJones** [comment]
I guess that merging would need to be done for context args and so on too, not just the string being internationalized.

```php
<?php _x( 'Post', 'Holds the
fence up', 'text-domain' ); ?>
```

**Comment by westonruter** [comment]
@GaryJones how's #545?

**Comment by GaryJones** [comment]
@westonruter #545 tested on my real and made-up code, and the error no longer appears. :+1:
